pkgs/tools/package-management/lix/common-lix.nix

```diff
@@ -91,23 +114,46 @@ stdenv.mkDerivation {
     ];
 
   strictDeps = true;
+  disallowedReferences = lib.optionals isLLVMOnly [
+    # We don't want the Clang.
+    stdenv.cc.cc
+    # We don't want the underlying GCC neither!
+    stdenv.cc.cc.stdenv.cc.cc
+  ];
+
+  # We only include CMake so that Meson can locate toml11, which only ships CMake dependency metadata.
+  dontUseCmakeConfigure = true;
 
   nativeBuildInputs =
     [
+      # python3.withPackages does not splice properly, see https://github.com/NixOS/nixpkgs/issues/305858
+      (python3.pythonOnBuildForHost.withPackages (p: [
+        p.pytest
+        p.pytest-xdist
+        p.python-frontmatter
+      ]))
       pkg-config
       flex
       jq
       meson
       ninja
       cmake
       python3
+      capnproto
```

Comment:
I don't remember if I were the one to introduce this originally but we need a very good reason for this addition, this is the root cause of the static build failures.

Reply:
Removed via the patch below.